pyacq/core/stream.py

```diff
@@ -285,84 +284,157 @@ class PlainDataSender:
     def __init__(self, socket, params):
         self.socket = socket
         self.params = params
-        self.copy = False
-        #self.copy = not self.params['protocol'] in ('inproc', 'tcp', 'ipc')
-        
         self.funcs = []
-        # compression
-        if self.params['compression'] == '':
-            self.funcs.append(self.to_bytes)
-        elif self.params['compression'] == 'blosc-blosclz':
-            # cname for ['blosclz', 'lz4', 'lz4hc', 'snappy', 'zlib']
-            self.funcs.append(self.compress_blosclz)
-        elif self.params['compression'] == 'blosc-lz4':
-            self.funcs.append(self.compress_blosclz4)        
-    
-    def to_bytes(self, index, data):
-        if not data.flags['C_CONTIGUOUS']:
-            # if not C continuous return transform to bytes
-            # so this is a copy
-            return index, data.tostring(order='C')
-        else:
-            # if C continuous let's keep  data as numpy in case of copy=True
-            # zmq shoudla avoid copy
-            return index, data
     
-    def compress_blosclz(self, index, data):
-        assert HAVE_BLOSC, "Cannot use blosclz compression; blosc package is not importable."
-        data = blosc.pack_array(data, cname='blosclz')
-        return index, data
-    
-    def compress_blosclz4(self, index, data):
-        assert HAVE_BLOSC, "Cannot use blosclz4 compression; blosc package is not importable."
-        data = blosc.pack_array(data, cname='lz4')
-        return index, data
-    
-    def send(self, index, data, autoswapaxes = True):
+    def send(self, index, data):
+        # optional pre-processing before send
         if isinstance(data, np.ndarray):
-            if autoswapaxes and self.params['timeaxis']!=0:
-                data = data.swapaxes(0, self.params['timeaxis'])
             for f in self.funcs:
                 index, data = f(index, data)
-        self.socket.send_multipart([np.int64(index), data], copy=self.copy)
+                
+        # serialize
+        dtype = data.dtype
+        shape = data.shape
+        buf, offset, strides = decompose_array(data)
+        
+        # compress
+        comp = self.params['compression']
+        if comp.startswith('blosc-'):
+            assert HAVE_BLOSC, "Cannot use blosclz4 compression; blosc package is not importable."
```

Comment:
This is buggy on my machine.
blosc need bytes for compression.
Could you also fix this in the PR ?